locality_alignment/backbones/dbot_clip.py

```diff
@@ -0,0 +1,696 @@
+        num_extra_tokens = model.pos_embed.shape[-2] - num_patches
+        # height (== width) for the checkpoint position embedding
+        orig_size = int((pos_embed_checkpoint.shape[-2] - num_extra_tokens) ** 0.5)
+        # height (== width) for the new position embedding
+        new_size = int(num_patches**0.5)
+        # class_token and dist_token are kept unchanged
+        if orig_size != new_size:
+            print("Position interpolate from %dx%d to %dx%d" % (orig_size, orig_size, new_size, new_size))
+            extra_tokens = pos_embed_checkpoint[:, :num_extra_tokens]
+            # only the position tokens are interpolated
+            pos_tokens = pos_embed_checkpoint[:, num_extra_tokens:]
+            pos_tokens = pos_tokens.reshape(-1, orig_size, orig_size, embedding_size).permute(0, 3, 1, 2)
+            pos_tokens = torch.nn.functional.interpolate(
+                pos_tokens, size=(new_size, new_size), mode="bicubic", align_corners=False
+            )
+            pos_tokens = pos_tokens.permute(0, 2, 3, 1).flatten(1, 2)
+            new_pos_embed = torch.cat((extra_tokens, pos_tokens), dim=1)
+            checkpoint_model["pos_embed"] = new_pos_embed
+
+    custom_load_state_dict(model, checkpoint_model, prefix=model_prefix)
+
+    return model
+
+
+def _create_dbot_vit(variant: str, pretrained: bool = False, **kwargs) -> dBOTCLIPVisionTransformer:
+    kwargs["pretrained_cfg"] = _cfg()
+
+    model = timm.models._builder.build_model_with_cfg(
+        dBOTCLIPVisionTransformer,
+        variant,
+        pretrained=False,
+        **kwargs,
+    )
+
+    if pretrained and variant.startswith("vit_base"):
+        # Load checkpoint.
+        cache_dir = torch.hub.get_dir()
+        file_name = "85.7_clip_base_pre.pth"
+        local_path = os.path.join(cache_dir, "checkpoints", file_name)
+
+        url = f"https://lf3-nlp-opensource.bytetos.com/obj/nlp-opensource/mmodal/dbot/{file_name}"
+        if os.path.exists(local_path):
+            logging.info(f"Loading {file_name} from local path: {local_path}")
+            checkpoint = torch.load(local_path, map_location=torch.device("cpu"))
+        else:
+            logging.info(f"Downloading {file_name} from url: {url}")
+            checkpoint = torch.hub.load_state_dict_from_url(url, map_location=torch.device("cpu"))
+
+        # dBOT modifications.
+        model = _load_dbot_clip_teacher(model, checkpoint)
+
+    else:
+        raise ValueError(f"Pretrained weights not available for {variant}")
+
+    return model
+
+
+@register_model
+def vit_base_patch16_dbot_clip_224(pretrained: bool = False, **kwargs) -> dBOTCLIPVisionTransformer:
+    model_args = dict(
+        patch_size=16,
+        embed_dim=768,
+        depth=12,
+        num_heads=12,
+        mlp_ratio=4,
+        qkv_bias=True,
+        norm_layer=partial(nn.LayerNorm, eps=1e-6),
+        init_values=0.1,  # 0.1 for base, 1e-5 for large. set 0 to disable layer scale
```

Comment:
Set to 0.1 and no comment. They're learnable values so it doesn't matter when we're loading a checkpoint